encoding/encoding.go

```diff
@@ -93,7 +93,7 @@ func (e Encoding) GetHealth() (Serializer, error) {
 	switch e {
 	case UnspecifiedEncoding, Thrift:
 		method, spec := getHealthSpec()
-		return thriftSerializer{method, spec}, nil
+		return &thriftSerializer{method, spec, defaultOpts}, nil
```

Comment:
Why make this a reference? `thriftSerializer` as a `Serializer` is still immutable since all methods are on the object, not the pointer.